### PR TITLE
Fix MAR records view breaking when deleting medOrders

### LINKED ISSUE #603: Clearing medications leaves MAR page broken
Create a set of medications
Create a few MARs
Return to the medications and clear the table.
Add some medications
Sometimes, if not always, the MAR page is broken and does not show the new medications.

Two part fix may be needed. May break this issue into two issues.

First, see that the MAR page works even if there are existing MAR records for medications that no longer exist in the system.  No need to show those records but do show them if it's easier to show them.

Second part, (new issue needed?)  When the Medications page (which is completely generated from definitions) is cleared of data then clear existing MAR records too.

Note, it may be easier to solve the second part which will also solve the first.  Or it might be easy to solve the first part.  Good luck. Neither seems easy.

### PR DESCRIPTION
**Pull request by viniciusarre**
Closes #603

### CODE REVIEW

File: client/src/inside/components/mar/mar-today.js [hunk 1]
```diff
@@ -83,7 +85,10 @@ export default class MarToday {
 
     // Match mars for the current data to the scheduled list of medications
     marRecsFiltered.forEach( mar =>  {
-      if(mar.scheduledTime ) {
+      const pk = aDaySchedule.find( pk => 
+        pk.hour24 === mar.scheduledTime
+      )
+      if(mar.scheduledTime && pk) {
```
Comment (bryan-gilbert) [inline client/src/inside/components/mar/mar-today.js]: Please add a comment to explain what's happening here.

File: client/src/inside/components/mar/mar-helper.js [hunk 2]
```diff
@@ -46,6 +46,18 @@ export default class MarHelper {
...
+    let asLoadedPageData = this.getEhrData_MarPageData()
+    
+    asLoadedPageData[marTableKey] = []
+    let payload = {
+      pageKey: MAR_PAGE_KEY,
+      value: asLoadedPageData
+    }
+    return this.ehrHelp._saveData(payload)
+  }
+
```
Comment (bryan-gilbert) [inline client/src/inside/components/mar/mar-helper.js]: nice

File: client/src/inside/components/page/EhrPageTable.vue [hunk 3]
```diff
@@ -84,6 +85,11 @@ export default {
     proceedClearAllData () {
       console.log('EhrPageTable clearAllData ', this.tableDef)
       this.ehrHelp.clearTable(this.tableKey)
+      if (this.pageDataKey === 'medicationOrders') {
+        const helper = new MarHelper(this.ehrHelp)
+        helper.clearAllData()
+      }
```
Comment (bryan-gilbert) [inline client/src/inside/components/page/EhrPageTable.vue]: Nice except I worry about string constants like 'medictionOrders' being used to identify the page.  Place a constant into the MarHelper with the page key and use that constant.  Look for other cases where 'medicationOrders' is used and replace with constant. Then comment the const to say its the page key for the medications orders page.

File: client/src/inside/components/mar/mar-today.js (hunk 1 above)
Comment (viniciusarre) [reply client/src/inside/components/mar/mar-today.js]: This is related to the issue's step one.

File: client/src/inside/components/page/EhrPageTable.vue (hunk 3 above)
Comment (viniciusarre) [reply client/src/inside/components/page/EhrPageTable.vue]: I have implemented this by reusing a constant which had been implemented in mar-helper (`MED_ORDERS_PAGE_KEY`) and implemented a different wrapper method in `mar-helper` called `triggerActionByPageKey`. Please, let me know your thoughts on it. Thank you!

**Pull request merged**